applications/IgaApplication/tests/cpp_tests/test_iga_shell_3p.cpp

```diff
@@ -0,0 +1,325 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License: BSD License
+//           Kratos default license: kratos/license.txt
+//
+
+// Project includes
+#include "containers/model.h"
+#include "testing/testing.h"
+
+#include "test_creation_utility.h"
+
+#include "custom_elements/shell_3p_element.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+    ///@name Type Definitions
+    ///@{
+
+    typedef std::size_t SizeType;
+    typedef std::size_t IndexType;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    typename Shell3pElement::Pointer GetShell3pElement(
+        ModelPart& rModelPart, SizeType PolynomialDegree, IntegrationPoint<3> IntegrationPoint)
+    {
+        // Set the element properties
+        auto p_elem_prop = rModelPart.CreateNewProperties(0);
+        p_elem_prop->SetValue(YOUNG_MODULUS, 200000000);
+        p_elem_prop->SetValue(POISSON_RATIO, 0.0);
+        p_elem_prop->SetValue(THICKNESS, 0.01);
+        const auto& r_clone_cl = KratosComponents<ConstitutiveLaw>::Get("LinearElasticPlaneStress2DLaw");
+        p_elem_prop->SetValue(CONSTITUTIVE_LAW, r_clone_cl.Clone());
+
+        auto p_quadrature_point = TestCreationUtility::GetQuadraturePointGeometry(
+            rModelPart, PolynomialDegree, IntegrationPoint);
+
+        return Kratos::make_intrusive<Shell3pElement>(1, p_quadrature_point, p_elem_prop);
+    }
+
+    // Tests the stiffness matrix of the Shell3pElement with a polynomial degree of p=3.
+    KRATOS_TEST_CASE_IN_SUITE(IgaShell3pElementP3, KratosIgaFastSuite)
+    {
+        Model current_model;
+        auto &r_model_part = current_model.CreateModelPart("ModelPart");
+        r_model_part.GetProcessInfo().SetValue(DOMAIN_SIZE, 3);
+
+        r_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+
+        IntegrationPoint<3> integration_point(0.0694318, 0.211324865405187, 0.0, 0.0869637);
+        auto p_shell_3p_element = GetShell3pElement(r_model_part, 3, integration_point);
+
+        TestCreationUtility::AddDisplacementDofs(r_model_part);
+
+        p_shell_3p_element->Initialize();
+
+        Matrix left_hand_side_matrix;
+        Vector right_hand_side_vector;
+        p_shell_3p_element->CalculateLocalSystem(left_hand_side_matrix, right_hand_side_vector, r_model_part.GetProcessInfo());
+
+        //Check RHS and LHS results
+        const double tolerance = 1.0e+0;
```

Comment:
Why do you use a different tolerance for this test case? (1 vs 0.1)

Reply:
Thanks for the finding. I am sorry for the mistake. I should use constant tolerance for each case. Please give me time to recheck and revise.

Reply:
No worries. I was trying it for 0.1 and was getting errors. Otherwise this looks really good. Thanks for the effort.

Reply:
Thanks. I have already fixed the errors. It is related to the inconsistency of the digits of the quadrature points & weights that I used in some cases.

Reply:
Awesome. I just found a not needed loop. Remove it and we can merge it